File: java/src/MoveZeros.java
public class MoveZeros {
    public void moveZeroes(int[] array) {
        int j = 0;
        for (int i = 0 ; i < array.length ; i++) {
            if (array[i] != 0) {
                array[j++] = array[i];
            }
        }
        while (j < array.length) {
            array[j++] = 0;
        }
    }
}

File: python/move_zeros.py
from typing import List


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        finalLength = 0
        index = 0
        while index < len(nums):
            if nums[index] != 0:
                nums[finalLength] = nums[index]
                finalLength += 1
            index += 1
        while finalLength < len(nums):
            nums[finalLength] = 0
            finalLength += 1
